components/nodemanager-service/nodemanager.go

```diff
@@ -133,3 +262,33 @@ func newGRPCServer(db *pgdb.DB, connFactory *secureconn.Factory, config *config.
 
 	return s
 }
+
+func createOrUpdateWorkflowSchedule(cerealManager *cereal.Manager, scheduleName string, jobName string, rule *rrule.RRule) error {
+	err := cerealManager.CreateWorkflowSchedule(context.Background(), scheduleName, jobName, nil, true, rule)
+
+	if err == nil {
```

Comment:
I have funny feelings about this early return, but I like less branching/indentation...